src/main/java/com/senkiv/bookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,129 @@
+package com.senkiv.bookstore.service.impl;
+
+import com.senkiv.bookstore.dto.OrderItemResponseDto;
+import com.senkiv.bookstore.dto.OrderRequestDto;
+import com.senkiv.bookstore.dto.OrderResponseDto;
+import com.senkiv.bookstore.dto.OrderStatusUpdateDto;
+import com.senkiv.bookstore.exception.OrderProcessingException;
+import com.senkiv.bookstore.mapper.OrderItemMapper;
+import com.senkiv.bookstore.mapper.OrderMapper;
+import com.senkiv.bookstore.model.Order;
+import com.senkiv.bookstore.model.OrderItem;
+import com.senkiv.bookstore.model.ShoppingCart;
+import com.senkiv.bookstore.model.User;
+import com.senkiv.bookstore.repository.OrderItemRepository;
+import com.senkiv.bookstore.repository.OrderRepository;
+import com.senkiv.bookstore.repository.ShoppingCartRepository;
+import com.senkiv.bookstore.service.OrderService;
+import com.senkiv.bookstore.service.ShoppingCartService;
+import jakarta.persistence.EntityNotFoundException;
+import jakarta.transaction.Transactional;
+import java.math.BigDecimal;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Transactional
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private static final String TOTAL_CALCULATION_ERR_MESSAGE =
+            "Error during total price calculation.";
+    private static final String NO_ORDER_FOR_USER_MESSAGE =
+            "User with id %s does not have order with id %s";
+    private static final String NO_ITEM_IN_ORDER_MESSAGE =
+            "There is no item with id %s in order with id %s";
+    private static final String EMPTY_CART_MESSAGE =
+            "Your cart is empty. Cannot place an order.";
+    private static final String NO_ORDER_WITH_SUCH_ID =
+            "There is no order with such id %s.";
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+    private final ShoppingCartService shoppingCartService;
+
+    public OrderResponseDto createOrder(Long userId, OrderRequestDto orderRequestDto) {
+        ShoppingCart shoppingCart = getShoppingCart(userId);
+        User owner = shoppingCart.getUser();
+        if (shoppingCartService.isCartEmpty(shoppingCart)) {
```

Comment:
why do you nee separate method for this logic? 
```suggestion
        if (shoppingCart.getCartItems().isEmpty()) {
```